### PR TITLE
Stereotypes for ItemFlow properties

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: #841

### What is the new behavior?

Add example for car braking system.

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by amolenaar** [comment]
@danyeaw 

Should stereotyped types of `itemProperty` show like this?

image

**Review by danyeaw — COMMENTED** [review]
Controller, Feedback, and Control Action are already defined in the RAAML profile under the STPA section. For the sysml-brake-system example, did we need to redefine them in a new profile?

File: gaphor/SysML/propertypages.py [hunk 1]
```diff
@@ -206,24 +215,43 @@ def _on_item_flow_active(self, switch, gparam):
             subject.informationFlow = iflow
             iflow.informationSource = subject.end[0].role
             iflow.informationTarget = subject.end[1].role
-            iflow.itemProperty = subject.model.create(sysml.Property)
+            iflow.itemProperty = subject.model.create(UML.Property)
```
Comment (danyeaw) [inline gaphor/SysML/propertypages.py:218]: Why the UML Property instead of the sysml one?

Comment (amolenaar) [reply gaphor/SysML/propertypages.py:218]: It's the same. `sysml` imports `Property` from `gaphor.UML`

**Comment by amolenaar** [comment]
> Controller, Feedback, and Control Action are already defined in the RAAML profile under the STPA section. For the sysml-brake-system example, did we need to redefine them in a new profile?

The default stereotypes are currently not picked up (they are only defined in the diagram item). Only the ones defined in the model. It would take a little more work to also display the default stereotypes (like `primitive` in the image above). It may be useful for other purposes to have a (single-dispatch) function that provides an optional stereotype name.

File: gaphor/SysML/propertypages.py (hunk 1 above)
Comment (danyeaw) [reply gaphor/SysML/propertypages.py:218]: Ya, I get that they are the same. It just seems clearer to me that the SysML profile uses SysML elements.

**Comment by danyeaw** [comment]
> The default stereotypes are currently not picked up (they are only defined in the diagram item). Only the ones defined in the model.

I am concerned if users have to redefine elements already in the RAAML profile, then we aren't really solving the original issue.

Do you think we should enhance that for at least Controller, Feedback, and Control Action? Should they only show when we have the RAAML profile selected?

**Comment by amolenaar** [comment]
> I am concerned if users have to redefine elements already in the RAAML profile, then we aren't really solving the original issue.

That would be weird, actually ;).

> Do you think we should enhance that for at least Controller, Feedback, and Control Action? 

We can do that, for starters. And come up with a more generic solution later, matching the "default" stereotype on classifiers.

> Should they only show when we have the RAAML profile selected?

I don't think we should do that. The profile is in no way related to the diagram.

**Comment by amolenaar** [comment]
Like this?

image

**Comment by danyeaw** [comment]
@amolenaar Yes, that looks great! How do I assign the Controller stereotype to the part autobrakeController and hydraulicController part properties?

**Comment by amolenaar** [comment]
We don't have those items in our toolbox currently. I think we should add them.

**Comment by amolenaar** [comment]
@danyeaw Is this ready for mainline?

**Pull request merged**